app/views/exam_templates/_boot.js.erb

```diff
@@ -14,6 +14,74 @@ document.addEventListener("DOMContentLoaded", function () {
   });
 });
 
+document.addEventListener("DOMContentLoaded", () => {
+  createConsumer().subscriptions.create(
+    {channel: "ExamTemplatesChannel", course_id: <%= @current_course.id %>},
+    {
+      received(data) {
+        if (data["status"] !== null) {
+          const message_data = generateMessage(data);
+          renderFlashMessages(message_data);
+          if (data["status"] === "completed") {
+            handleJobCompletion(data);
+          }
+        }
+      },
+      connected() {},
+      disconnected() {},
+    }
+  );
+});
+
+function generateMessage(status_data) {
+  let message_data = {};
+  switch (status_data["status"]) {
+    case "queued":
+      message_data["notice"] = I18n.t("poll_job.queued");
+      break;
+    case "in_progress":
+      if (status_data["job_class"] === "SplitPdfJob") {
+        message_data["notice"] = I18n.t("poll_job.split_pdf_job", {
```

Comment:
Because both the `status_data` fields and the message attributes are the same in both branches, you can simplify this if statement to remove duplicate code. The value that is dependent on the if branches is the first argument to `I18n.t`; everything else is the same, and can be done after the if statement.

Reply:
@hemmatio I see, yes this is a bit annoying. It's fine to keep the duplicated `I18n.t` call. However, you can still define a single js object to store `{ progress: progress, ...}` and pass that into the `I18n.t` call, so you reduce the duplication of the argument passing.